File: src/api/currencies.ts
import api from "../lib/axios";

export interface Currency {
  char_code: string;
  nominal: string;
  name: string;
  value: number;
  unit_rate: number;
}

export const fetchCurrencies = async (): Promise<Currency[]> => {
  try {
    const response = await api.get<Currency[]>('/currencies/');
    return response.data;
  } catch (error) {
    console.error("Ошибка при загрузке валют:", error);
    return [];
  }
};

File: app/schemas/currencies.py
from pydantic import BaseModel, Field


class GetCurrencies(BaseModel):
    char_code: str = Field(description='Код валюты')
    nominal: int = Field(description='Количество валюты')
    name: str = Field(description='Название валюты')
    value: float = Field(description='Стоимость в рублях за nominal валюты')
    unit_rate: float = Field(description='Стоимость в рублях за 1 валюты')
